module/VuFindConsole/src/VuFindConsole/Command/Util/BrowscapCommand.php

```diff
@@ -0,0 +1,193 @@
+     * Configure the command.
+     *
+     * @return void
+     */
+    protected function configure()
+    {
+        $this
+            ->setDescription('Browscap Cache Manager')
+            ->setHelp('Manages the browscap cache.')
+            ->addArgument(
+                'function',
+                InputArgument::REQUIRED,
+                'Function to execute. Currently the only supported function is: update'
+            )
+            ->addOption(
+                'file-type',
+                null,
+                InputOption::VALUE_REQUIRED,
+                'Browscap file type (default, lite or full)',
+                'normal'
+            );
+    }
+
+    /**
+     * Run the command.
+     *
+     * @param InputInterface  $input  Input object
+     * @param OutputInterface $output Output object
+     *
+     * @return int 0 for success
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        if ($input->getArgument('function') !== 'update') {
+            $output->writeln('<error>Invalid function specified</error>');
+            return Command::FAILURE;
+        }
+        switch ($input->getOption('file-type')) {
+            case 'full':
+                $type = \BrowscapPHP\Helper\IniLoaderInterface::PHP_INI_FULL;
+                break;
+            case 'lite':
+                $type = \BrowscapPHP\Helper\IniLoaderInterface::PHP_INI_LITE;
+                break;
+            case 'normal':
+                $type = \BrowscapPHP\Helper\IniLoaderInterface::PHP_INI;
+                break;
+            default:
+                $output->writeln('<error>Invalid file-type specified</error>');
+                return Command::FAILURE;
+        }
+
+        $cache = new SimpleCacheDecorator($this->cacheManager->getCache('browscap'));
+        $logger = new ConsoleLogger($output, [LogLevel::INFO => OutputInterface::VERBOSITY_NORMAL]);
+        $client = new \GuzzleHttp\Client($this->getGuzzleConfig());
+
+        $bc = new \BrowscapPHP\BrowscapUpdater($cache, $logger, $client);
+        $logger->info('Checking for update...');
+        try {
+            $bc->checkUpdate();
+        } catch (\BrowscapPHP\Exception\NoNewVersionException $e) {
+            $logger->info('No newer version available.');
+            return Command::SUCCESS;
+        } catch (\BrowscapPHP\Exception\NoCachedVersionException $e) {
+            $logger->info('No cached version available.');
+        } catch (\Exception $e) {
+            // Output the exception and continue (assume we don't have a current version):
+            $logger->warning((string)$e);
+        }
+        $logger->info('Updating browscap cache...');
+        $bc->update(\BrowscapPHP\Helper\IniLoaderInterface::PHP_INI_FULL);
+        $logger->info('Update complete.');
+
+        return Command::SUCCESS;
+    }
+
+    /**
+     * Get Guzzle options
+     *
+     * @return array
+     */
+    protected function getGuzzleConfig(): array
```

Comment:
Should we register the Guzzle client as a service instead of embedding the logic here? It seems like it would be useful to have a reusable mechanism for obtaining a Guzzle client, and this could be a first small step toward migrating away from the legacy \Laminas\Http client.

Reply:
Done. Our proxy mechanism is a bit complicated, but this works.